Gems/Atom/Tools/MaterialCanvas/Code/Source/MaterialCanvasApplication.cpp

```diff
@@ -218,4 +222,45 @@ namespace MaterialCanvas
     {
         return m_window.get();
     }
+
+    void MaterialCanvasApplication::ApplyShaderBuildSettings()
+    {
+        // If minimal shader build settings are enabled, copy a settings registry file stub into the user settings folder. This will
+        // override AP and shader build settings, disabling shaders and shader variant building for inactive platforms and RHI. Copying any
+        // of these settings files requires restarting the application and the asset processor for the changes to be picked up.
+        if (auto fileIO = AZ::IO::FileIOBase::GetInstance())
+        {
+            const AZ::IO::Path settingsPath(
+                AZStd::string::format("%s/user/Registry/user_minimal_shader_build.setreg", AZ::Utils::GetProjectPath().c_str()));
+            const AZ::IO::Path settingsPathStub(
+                AtomToolsFramework::GetPathWithoutAlias("@gemroot:MaterialCanvas@/Registry/user_minimal_shader_build.setregstub"));
+            const AZ::IO::Path settingsPathDx12(
+                AZStd::string::format("%s/user/Registry/user_minimal_shader_build_dx12.setreg", AZ::Utils::GetProjectPath().c_str()));
+            const AZ::IO::Path settingsPathDx12Stub(
+                AtomToolsFramework::GetPathWithoutAlias("@gemroot:MaterialCanvas@/Registry/user_minimal_shader_build_dx12.setregstub"));
+            const bool enableMinimalShaderBuilds =
+                AtomToolsFramework::GetSettingsValue<bool>("/O3DE/Atom/Tools/MaterialCanvas/EnableMinimalShaderBuilds", false);
```

Comment:
Couple of nits here, but neither are blockers.
1) I wish we had a more generic way to get the user folder (@user@ perhaps). 
2) This makes it seem like vulkan is the default windows platform. Not sure if the default should be dx12 and vulkan the callout

Reply:
We can just add a GetProjectUserPath function to [Utils.h](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Utils/Utils.h#L110-L113).

It isn't actually that much code
```c++
    AZ::IO::FixedMaxPathString GetProjectPath(AZ::SettingsRegistryInterface* settingsRegistry)
    {
        if (settingsRegistry == nullptr)
        {
            settingsRegistry = AZ::SettingsRegistry::Get();
        }


        if (settingsRegistry != nullptr)
        {
            if (AZ::IO::FixedMaxPathString settingsValue;
                settingsRegistry->Get(settingsValue, AZ::SettingsRegistryMergeUtils::FilePathKey_ProjectUserPath))
            {
                return settingsValue;
            }
        }
        return {};
    }
```

Reply:
Also FYI, we do have a [Utils::GetGemPath](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Utils/Utils.h#L89-L93) function that can avoid needing to access the FileIO base for resolving a gem paht

Reply:
As discussed offline, the path should resolve to the same thing either way. But using the user alias allows flexibility in case the user folder ever gets moved somewhere else.

In this case, DX12 is the exception, not the default case. Currently, every other platform only has one RHI back end. Windows has Vulkan and DX12. I need additional messaging if the setting is enabled but you switched between DX12 and Vulkan, which would require restart because the settings would silently change.

Ultimately, we need to get to a point where we can avoid using the AP for generating the things we want to preview the material canvas RFC mentions, and there have been other discussions about, exposing APIs from the shader builder to allow direct control of what gets built, and invoking it, directly from material canvas or any other tool without going through the AP.